### ISSUE TITLE
Missing collection item when they are not wrapped during unmarshal with multiple namespaces

### ISSUE THREAD

**Issue by wanchongtai** [FasterXML/jackson-dataformat-xml#307]
1. I have the following xml to unmarshal 

```xml
<?xml version="1.0" encoding="utf-8"?>
<customer xmlns="http://www.archer-tech.com/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    <customerId>1</customerId>
    <customerName>Michael Judy</customerName>
    <account>
        <accountId>100</accountId>
        <accountName>Michael</accountName>
        <postcode xsi:nil="true"></postcode>
    </account>
    <account>
        <accountId>200</accountId>
        <accountName>Judy</accountName>
        <postcode xsi:nil="true"></postcode>
    </account>	
</customer>
```

2. domains

```java
@JacksonXmlRootElement(localName = "customer")
public class CustomerWithoutWrapper {
    Long customerId;
    String customerName;

    @JacksonXmlElementWrapper(useWrapping = false)
    List<Account> account = new ArrayList<>();
...setter/getter...
}

public class Account {
    Long accountId;
    String accountName;
    String postcode;
...setter/getter...
}
```

3. unmarshal process

```java
        ObjectMapper objectMapper = new XmlMapper();
        CustomerWithoutWrapper customerWithOutWrapper =
                objectMapper.readValue(getCustomerWithoutWrapperXml(), CustomerWithoutWrapper.class);
```

4. result

there is not exception, however only the last account item is kept. the first account item is missing from account collection after unmarshalling 

incorrect result 

```
CustomerWithoutWrapper{customerId=1, customerName='Michael Judy', account=[Account{accountId=200, accountName='Judy', postcode=''}]}
```

5. however when account items are wrapped as following, and change the domain mapping accordingly, then Jackson can unmarshall xml correctly.
e.g. 

```xml
<?xml version="1.0" encoding="utf-8"?>
<customer xmlns="http://www.archer-tech.com/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
	<customerId>1</customerId>
	<customerName>Michael Judy</customerName>
	<accounts>
		<account>
			<accountId>100</accountId>
			<accountName>Michael</accountName>
			<postcode xsi:nil="true"></postcode>
		</account>
		<account>
			<accountId>200</accountId>
			<accountName>Judy</accountName>
			<postcode xsi:nil="true"></postcode>
		</account>	
	</accounts>
</customer>
```

```java
@JacksonXmlRootElement(localName = "customer")
public class CustomerWithWrapper {
    Long customerId;

    String customerName;

    @JacksonXmlElementWrapper(localName = "accounts")
    List<Account> account = new ArrayList<>();
...
}
```

result - correct when accounts are wrapped.

```
CustomerWithWrapper{customerId=1, customerName='Michael Judy', account=[Account{accountId=100, accountName='Michael', postcode=''}, Account{accountId=200, accountName='Judy', postcode=''}]}
```

**Comment by wanchongtai** [comment]
re-attach xml file content
====
<?xml version="1.0" encoding="utf-8"?>
<customer xmlns="http://www.archer-tech.com/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
	<customerId>1</customerId>
	<customerName>Michael Judy</customerName>
		<account>
			<accountId>100</accountId>
			<accountName>Michael</accountName>
			<postcode xsi:nil="true"></postcode>
		</account>
		<account>
			<accountId>200</accountId>
			<accountName>Judy</accountName>
			<postcode xsi:nil="true"></postcode>
		</account>	
</customer>
====

**Comment by cowtowncoder** [comment]
This will be fixed in 2.12.0; added a test to verify; test failed for 2.11 branch.
